### PR TITLE
Render actor entity in animViewportRenderer

### PR DESCRIPTION
**Pull request by amzn-rhhong**

### CODE REVIEW

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp [hunk 1]
```diff
@@ -238,6 +227,75 @@ namespace EMStudio
             m_cameraEntity->GetId(), &AZ::Debug::CameraControllerRequestBus::Events::Enable, azrtti_typeid<AZ::Debug::NoClipControllerComponent>());
     }
 
+    void AnimViewportRenderer::ReinitActorEntities()
+    {
+        // 1. Destroy all the entities that does not point to any actorAsset anymore.
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp:232]: "all the entities that does not" -> that do not

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp [hunk 2]
```diff
@@ -30,12 +32,15 @@
...
+#include <EMotionFX/Source/EMotionFXManager.h>
+#include <EMotionFX/Source/ActorManager.h>
 #include <EMStudio/AnimViewportRenderer.h>
-#pragma optimize("", off)
+
 
 namespace EMStudio
 {
     static constexpr float DepthNear = 0.01f;
+    static constexpr const char* const s_actorComponentTypeId = "{BDC97E7F-A054-448B-A26F-EA2B5D78E377}";
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp:43]: We seem to have access to the actor component, can you remove this and just use the rtti system here and get it from the real class?

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h [hunk 3]
```diff
@@ -46,9 +47,21 @@ namespace EMStudio
         AnimViewportRenderer(AZStd::shared_ptr<AZ::RPI::WindowContext> windowContext);
         ~AnimViewportRenderer();
 
+        void Reinit();
+
     private:
 
-        void Reset();
+        // This function reset the light, camera and other environment settings.
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h:54]: reset -> resets

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h [hunk 4]
```diff
@@ -46,9 +47,21 @@ namespace EMStudio
...
+        void ResetEnvironment();
+
+        // This function create in-editor entities for all the actor asset stored in the actor manager,
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h:57]: create -> creates
the actor asset -> actor assets

Comment (amzn-jillich) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h:57]: delete -> deletes etc.

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h [hunk 5]
```diff
@@ -68,6 +81,7 @@ namespace EMStudio
         AZ::Entity* m_modelEntity = nullptr;
         AZ::Data::AssetId m_modelAssetId;
         AZ::Entity* m_gridEntity = nullptr;
+        AZStd::vector<AZ::Entity*> m_actorEntities;
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h:84]: How often is the FindActorEntity() used at runtime? Would it make sense to use a set here?

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.h [hunk 6]
```diff
@@ -18,6 +19,7 @@ namespace EMStudio
     {
     public:
         AnimViewportWidget(QWidget* parent = nullptr);
+        AnimViewportRenderer* GetAnimViewportRenderer() { return m_renderer.get();}
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.h:22]: Nit: Space missing between ; and }

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.cpp [hunk 7]
```diff
@@ -75,9 +85,40 @@ namespace EMStudio
         verticalLayout->setSizeConstraint(QLayout::SetNoConstraint);
         verticalLayout->setSpacing(1);
         verticalLayout->setMargin(0);
-
         m_animViewportWidget = new AnimViewportWidget(m_innerWidget);
 
+        // Register command callbacks.
+        m_createActorInstanceCallback = new CreateActorInstanceCallback(false);
+        EMStudioManager::GetInstance()->GetCommandManager()->RegisterCommandCallback("CreateActorInstance", m_createActorInstanceCallback);
```
Comment (amzn-jillich) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AtomRenderPlugin.cpp:92]: What about the remove actor instance one?

File: Gems/EMotionFX/Code/Include/Integration/ActorComponentBus.h [hunk 8]
```diff
@@ -95,6 +95,10 @@ namespace EMotionFX
             /// Returns skinning method used by the actor.
             virtual SkinningMethod GetSkinningMethod() const = 0;
 
+            // Use this to alter the actor asset.
+            virtual void SetActorAsset(AZ::Data::Asset<EMotionFX::Integration::ActorAsset> actorAsset) = 0;
+            // virtual AZ::Data::Asset<const EMotionFX::Integration::ActorAsset> GetAsset() const = 0;
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/Include/Integration/ActorComponentBus.h:100]: Remove?

**Review by amzn-jillich — APPROVED** [review]

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h (hunk 5 above)
Comment (amzn-rhhong) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.h:84]: Very rarely. We do not use multiple actors often anyway.

**Review by lsemp3d — APPROVED** [review]

**Pull request merged**